### ISSUE TITLE
Support for self-hosted instances

### ISSUE THREAD

**Issue by cloneofghosts** [Pirate-Weather/pirate-weather-ha#358]
### Describe the feature

Not sure this is possible but it would be nice if the integration could connect to self-hosted instances of the API rather than the production API. Maybe have an option for production, development and self-hosted?

### Home Assistant version

2024.12.3

### Integration version

1.6.3

### Acknowledgements

- [X] I have searched this repository and [API Repository](https://github.com/Pirate-Weather/pirateweather/) to see if the feature has already been requested.
- [X] This featre request is sepecific to the HA Integration.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
@alexander0042 Was thinking of setting this up this weekend but am unsure how to add the option to the config flow. Is it pretty simple or is it more complicated than it seems?

**Comment by alexander0042** [comment]
Thanks for reminding me about this! It really isn't too bad, and I'll see if I can give it an update tomorrow. It's mostly just editing the config_flow.py file to add a URL variable that defaults to `https://api.pirateweather.net/forecast`, then replacing instances in the code where the URL is hardcoded in with the variable. Would also make it possible to hit the dev endpoint, which is handy

**Comment by cloneofghosts** [comment]
Yeah, the plan was to have it setup to select live (default), dev and the self-host url. Should probably add some text saying that if you're not sure leave it as the live URL since I know people might get confused.

**Comment by cloneofghosts** [comment]
@alexander0042 Were you able to look at it last week or do you think you'll be able to look at it this week?

**Comment by alexander0042** [comment]
I actually did! Just waiting on some tests, but should have a PR soon!

**Comment by cloneofghosts** [comment]
I don't have my devcontainer setup at the moment (have a new computer and haven't gotten around to setting it up) but I can set it up and test once you create a PR if you want.

**Comment by alexdelprete** [comment]
Any docs on how to self-host the API?
